src/fsharp/TastOps.fs

```diff
@@ -5951,12 +5951,12 @@ let isRecdOrStructTyconRefAssumedImmutable (g: TcGlobals) (tcref: TyconRef) =
 
 let isTyconRefReadOnly g m (tcref: TyconRef) =
     tcref.CanDeref &&
-    match tcref.TryIsReadOnly with 
-    | ValueSome res -> res
-    | _ ->
+    ((match tcref.TryIsReadOnly with 
```

Comment:
tbh instead of multiple parens - can we rewrite it as nested ifs?

Reply:
Didn't like the parens either. It's just what I did. Should be better now